src/Catalyst.Core.Lib.Tests/UnitTests/Rpc/IO/Observers/GetPeerInfoRequestObserverTests.cs

```diff
@@ -0,0 +1,194 @@
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using Catalyst.Abstractions.IO.Messaging.Dto;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Network;
+using Catalyst.Core.Lib.P2P.Models;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Modules.Rpc.Server.IO.Observers;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Wire;
+using Catalyst.Protocol.Rpc.Node;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using FluentAssertions;
+using Microsoft.Reactive.Testing;
+using NSubstitute;
+using Serilog;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.Rpc.IO.Observers
+{
+    /// <summary>
+    ///     Tests the get peer info calls
+    /// </summary>
+    public sealed class GetPeerInfoRequestObserverTests
+    {
+        private readonly ILogger _logger;
+        private readonly IChannelHandlerContext _fakeContext;
+        private readonly IPeerRepository _peerRepository;
+
+        public GetPeerInfoRequestObserverTests()
+        {
+            _logger = Substitute.For<ILogger>();
+            _fakeContext = Substitute.For<IChannelHandlerContext>();
+
+            var fakeChannel = Substitute.For<IChannel>();
+            _fakeContext.Channel.Returns(fakeChannel);
+
+            var peers = GetPeerTestData();
+
+            _peerRepository = Substitute.For<IPeerRepository>();
+            _peerRepository.FindAll(Arg.Any<Expression<Func<Peer, bool>>>())
+               .Returns(ci => { return peers.Where(p => ((Expression<Func<Peer, bool>>) ci[0]).Compile()(p)); });
+        }
+
+        public IEnumerable<Peer> GetPeerTestData()
+        {
+            yield return new Peer
+            {
+                PeerId =
+                    PeerIdHelper.GetPeerId("publickey-1", IPAddress.Parse("172.0.0.1"), 9090),
+                LastSeen = DateTime.UtcNow, Created = DateTime.UtcNow
+            };
+            yield return new Peer
+            {
+                PeerId =
+                    PeerIdHelper.GetPeerId("publickey-2", IPAddress.Parse("172.0.0.2"), 9090),
+                LastSeen = DateTime.UtcNow, Created = DateTime.UtcNow
+            };
+            yield return new Peer
+            {
+                PeerId =
+                    PeerIdHelper.GetPeerId("publickey-3", IPAddress.Parse("172.0.0.3"), 9090),
+                LastSeen = DateTime.UtcNow, Created = DateTime.UtcNow
+            };
+            yield return new Peer
+            {
+                PeerId =
+                    PeerIdHelper.GetPeerId("publickey-3", IPAddress.Parse("172.0.0.3"), 9090),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure using this hardcoded IP address '172.0.0.3' is safe here.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)